gobblin-runtime/src/main/java/org/apache/gobblin/runtime/GobblinMultiTaskAttempt.java

```diff
@@ -160,25 +163,40 @@ public void run()
       }
     } catch (InterruptedException interrupt) {
       log.info("Job interrupted by InterrupedException.");
-      interruptTaskExecution(countDownLatch);
+      interruptTaskExecution(Optional.of(countDownLatch));
     }
     log.info("All assigned tasks of job {} have completed in container {}", jobId, containerIdOptional.or(""));
   }
 
-  private void interruptTaskExecution(CountDownLatch countDownLatch)
-      throws InterruptedException {
+  /**
+   * A helper function that shutting down all outstanding tasks and destroy taskExecutor if timeout on waiting for
```

Comment:
Change to "...that shuts down all outstanding tasks and shuts down the taskExecutor if it times out on a task termination"